### PR TITLE
[full-ci][tests-only] Remove @yetToImplement tags from tests (Part-2)

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
remove @yetToImplement from tests and add them to expected-failures if required features not implemented yet

## Related Issue
- part of https://github.com/owncloud/web/issues/4977

## How Has This Been Tested?
- test environment: local

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/webUISharingInternalGroupsToRoot/shareWithGroups.feature [hunk 1]
```diff
@@ -147,22 +147,6 @@ Feature: Sharing files and folders with internal groups
...
-    Then user "Alice" should not be able to share file "lorem.txt" with group "system-group" using the sharing API
-
-  @skip @yetToImplement
-  Scenario: user tries to share a folder in a group which is excluded from receiving share
-    Given user "Alice" has created folder "simple-folder"
-    And group "system-group" has been created
-    And the administrator has browsed to the admin sharing settings page
-    When the administrator excludes group "system-group" from receiving shares using the webUI
-    Then user "Alice" should not be able to share folder "simple-folder" with group "system-group" using the sharing API
-
```
Comment (saw-jan) [inline tests/acceptance/features/webUISharingInternalGroupsToRoot/shareWithGroups.feature:165]: Should cover by https://github.com/owncloud/web/blob/86f35b6018803295aa0b00b23dbbd8de1ac49a54/tests/acceptance/features/webUISharingInternalGroupsToRoot/shareWithGroups.feature#L197

File: tests/acceptance/features/webUISharingInternalUsersBlacklisted/shareWithUsers.feature [hunk 2]
```diff
@@ -12,16 +12,6 @@ Feature: Sharing files and folders with internal users
...
 
-  @skip @yetToImplement
-  Scenario: user tries to share a file from a group which is blacklisted from sharing
-    Given group "grp1" has been created
-    And user "Alice" has been added to group "grp1"
-    And user "Carol" has been created with default attributes
-    And the administrator has browsed to the admin sharing settings page
-    When the administrator enables exclude groups from sharing using the webUI
-    And the administrator adds group "grp1" to the group sharing blacklist using the webUI
-    Then user "Alice" should not be able to share file "testimage.jpg" with user "Carol" using the sharing API
```
Comment (saw-jan) [inline tests/acceptance/features/webUISharingInternalUsersBlacklisted/shareWithUsers.feature:23]: very similar to https://github.com/owncloud/web/blob/86f35b6018803295aa0b00b23dbbd8de1ac49a54/tests/acceptance/features/webUISharingInternalUsersBlacklisted/shareWithUsers.feature#L80

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>SharingInternalGroupsToRoot</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/16104/

<p>


### webUISharingInternalGroupsToRootEdgeCases-shareWithGroupsEdgeCases-feature-36.png
  

### webUISharingInternalGroupsToRootEdgeCases-shareWithGroupsEdgeCases-feature-37.png
  


</p></details>

File: tests/acceptance/features/webUISharingInternalGroupsToRootEdgeCases/shareWithGroupsEdgeCases.feature [hunk 3]
```diff
@@ -65,39 +63,39 @@ Feature: Sharing files and folders with internal groups
...
-    When the user shares folder "simple-folder" with user "Alice Hansen" as "Editor" using the webUI
-    And the user shares folder "simple-folder" with group "User1" as "Editor" using the webUI
-    And the user re-logs in as "Alice" using the webUI
-    Then folder "simple-folder" should be listed on the webUI
-#    And folder "simple-folder" should be marked as shared by "Carol King" on the webUI
-    When the user re-logs in as "Brian" using the webUI
-    Then folder "simple-folder" should be listed on the webUI
-#    And folder "simple-folder" should be marked as shared with "User1" by "Carol King" on the webUI
+    And the user shares file "randomfile.txt" with user "Alice Hansen" as "Editor" using the webUI
+    When the user shares file "randomfile.txt" with group "ALICE" as "Editor" using the webUI
```
Comment (Talank) [inline tests/acceptance/features/webUISharingInternalGroupsToRootEdgeCases/shareWithGroupsEdgeCases.feature:80]:
```suggestion
    When the user shares file "randomfile.txt" with user "Alice Hansen" as "Editor" using the webUI
    And the user shares file "randomfile.txt" with group "ALICE" as "Editor" using the webUI
```
Looks like this should be in a when step.

File: tests/acceptance/features/webUISharingNotifications/notificationLink.feature [hunk 4]
```diff
@@ -6,16 +6,20 @@ Feature: Display notifications when receiving a share and follow embedded links
...
-  @smokeTest @skip @yetToImplement
+  @smokeTest @issue-5227 @issue-ocis-14
   Scenario: notification link redirection in case a share is pending
     Given user "Alice" has created folder "simple-folder"
-    And the setting "Automatically accept new incoming local user shares" in the section "Sharing" has been disabled
     And user "Alice" has shared folder "/simple-folder" with user "Brian"
-    When the user follows the link of the first notification on the webUI
-    Then the user should be redirected to a webUI page with the title "Shared with you - %productname%"
+    When the user re-logs in as "Brian" using the webUI
+    When the user follows the link of following share from notification using the webUI
```
Comment (Talank) [inline tests/acceptance/features/webUISharingNotifications/notificationLink.feature:22]:
```suggestion
    When the user re-logs in as "Brian" using the webUI
    And the user follows the link of following share from notification using the webUI
```

File: tests/acceptance/features/webUISharingNotificationsToRoot/notificationLink.feature [hunk 5]
```diff
@@ -6,16 +6,19 @@ Feature: Display notifications when receiving a share and follow embedded links
...
-  @smokeTest @skip @yetToImplement
+  @smokeTest @issue-5227
   Scenario: notification link redirection in case a share is pending
     Given user "Alice" has created folder "simple-folder"
-    And the setting "Automatically accept new incoming local user shares" in the section "Sharing" has been disabled
     And user "Alice" has shared folder "/simple-folder" with user "Brian"
-    When the user follows the link of the first notification on the webUI
-    Then the user should be redirected to a webUI page with the title "Shared with you - %productname%"
+    When the user re-logs in as "Brian" using the webUI
+    When the user follows the link of following share from notification using the webUI
```
Comment (Talank) [inline tests/acceptance/features/webUISharingNotificationsToRoot/notificationLink.feature:21]:
```suggestion
    When the user re-logs in as "Brian" using the webUI
    And the user follows the link of following share from notification using the webUI
```

**Review by Talank — COMMENTED** [review]
LGTM

File: tests/acceptance/features/webUISharingNotificationsToRoot/notificationLink.feature [hunk 6]
```diff
@@ -6,16 +6,19 @@ Feature: Display notifications when receiving a share and follow embedded links
...
     Given user "Alice" has created folder "simple-folder"
-    And the setting "Automatically accept new incoming local user shares" in the section "Sharing" has been disabled
     And user "Alice" has shared folder "/simple-folder" with user "Brian"
-    When the user follows the link of the first notification on the webUI
-    Then the user should be redirected to a webUI page with the title "Shared with you - %productname%"
+    When the user re-logs in as "Brian" using the webUI
+    And the user follows the link of following share from notification using the webUI
+      | resource | simple-folder |
+      | sharer   | Alice         |
+    # Then the user should be redirected to a webUI page with the title "Shared with you - %productname%"
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingNotificationsToRoot/notificationLink.feature:24]: why is this step commented out?

File: tests/acceptance/features/webUISharingNotifications/notificationLink.feature [hunk 7]
```diff
@@ -6,16 +6,20 @@ Feature: Display notifications when receiving a share and follow embedded links
...
     Given user "Alice" has created folder "simple-folder"
-    And the setting "Automatically accept new incoming local user shares" in the section "Sharing" has been disabled
     And user "Alice" has shared folder "/simple-folder" with user "Brian"
-    When the user follows the link of the first notification on the webUI
-    Then the user should be redirected to a webUI page with the title "Shared with you - %productname%"
+    When the user re-logs in as "Brian" using the webUI
+    And the user follows the link of following share from notification using the webUI
+      | resource | simple-folder |
+      | sharer   | Alice         |
+    # Then the user should be redirected to a webUI page with the title "Shared with you - %productname%"
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingNotifications/notificationLink.feature:25]: why is this step commented out?

File: tests/acceptance/features/webUISharingInternalUsersToRootBlacklisted/shareWithUsers.feature [hunk 8]
```diff
@@ -10,17 +10,6 @@ Feature: Sharing files and folders with internal users
       | Alice    |
       | Brian    |
 
-  @skip @yetToImplement
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingInternalUsersToRootBlacklisted/shareWithUsers.feature:13]: why deleting this test?

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 9]
```diff
@@ -162,22 +162,6 @@ Feature: Sharing files and folders with internal groups
     Then file "lorem.txt" should be listed on the webUI
     And as "Carol" the content of "new-simple-folder/lorem.txt" should be the same as the original "simple-folder/lorem.txt"
 
-  @issue-5015
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature:165]: why deleting these tests?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/webUISharingInternalGroupsEdgeCases/shareWithGroupsEdgeCases.feature [hunk 10]
```diff
@@ -77,49 +72,43 @@ Feature: Sharing files and folders with internal groups
...
-#    And folder "simple-folder (2)" should be marked as shared with "User1" by "Carol King" on the webUI
+    When the user shares file "randomfile.txt" with user "Alice Hansen" as "Editor" using the webUI
+    And the user shares file "randomfile.txt" with group "ALICE" as "Editor" using the webUI
+    And user "Alice" accepts the share "randomfile.txt" offered by user "Carol" using the sharing API
+    And user "Brian" accepts the share "randomfile.txt" offered by user "Carol" using the sharing API
+    And the user opens the share creation dialog in the webUI
+    And the user types "ALICE" in the share-with-field
+    Then "group" "ALICE" should not be listed in the autocomplete list on the webUI
+    And the content of file "Shares/randomfile.txt" for user "Brian" should be "Carol file"
+    And the content of file "Shares/randomfile.txt" for user "Alice" should be "Carol file"
```
Comment (jasson99) [inline tests/acceptance/features/webUISharingInternalGroupsEdgeCases/shareWithGroupsEdgeCases.feature:96]: Do we need to assert these as well here?

File: tests/acceptance/features/webUISharingInternalGroupsEdgeCases/shareWithGroupsEdgeCases.feature [hunk 11]
```diff
@@ -77,49 +72,43 @@ Feature: Sharing files and folders with internal groups
...
-#    Then folder "simple-folder" should be marked as shared with "User1" by "Carol King" on the webUI
+    When the user shares file "randomfile.txt" with group "ALICE" as "Editor" using the webUI
+    And the user shares file "randomfile.txt" with user "Alice Hansen" as "Editor" using the webUI
+    And user "Alice" accepts the share "randomfile.txt" offered by user "Carol" using the sharing API
+    And user "Brian" accepts the share "randomfile.txt" offered by user "Carol" using the sharing API
+    And the user opens the share creation dialog in the webUI
+    And the user types "Alice" in the share-with-field
+    Then "user" "Alice Hansen" should not be listed in the autocomplete list on the webUI
+    And the content of file "Shares/randomfile.txt" for user "Brian" should be "Carol file"
+    And the content of file "Shares/randomfile.txt" for user "Alice" should be "Carol file"
```
Comment (jasson99) [inline tests/acceptance/features/webUISharingInternalGroupsEdgeCases/shareWithGroupsEdgeCases.feature:114]: Is this scenario clear? I could not understand the sequence. When the user opens the share creation dialog in the webUI is for user `Carol` isn't it? And for which resource is the share creation dialog opened? Also. why shouldnot `Alice` hansen be listed in the autocomplete list? Do we expect to be listed anything?

Comment (jasson99) [reply tests/acceptance/features/webUISharingInternalGroupsEdgeCases/shareWithGroupsEdgeCases.feature:114]: same in the above scenario.

Comment (saw-jan) [reply tests/acceptance/features/webUISharingInternalGroupsEdgeCases/shareWithGroupsEdgeCases.feature:114]: The scenario is to:
- share file with group `ALICE`
- share file with user `Alice`
- and then check if I can share again with `Alice` using webUI

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature (hunk 9 above)
Comment (saw-jan) [reply tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature:165]: should cover by https://github.com/owncloud/web/blob/e1b958326be41975cd53445a6d88bcf92f3be1e2/tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature#L200-L209

File: tests/acceptance/features/webUISharingInternalGroupsEdgeCases/shareWithGroupsEdgeCases.feature (hunk 10 above)
Comment (saw-jan) [reply tests/acceptance/features/webUISharingInternalGroupsEdgeCases/shareWithGroupsEdgeCases.feature:96]: May be we don't.
Any suggestions?

File: tests/acceptance/features/webUISharingInternalUsersToRootBlacklisted/shareWithUsers.feature (hunk 8 above)
Comment (saw-jan) [reply tests/acceptance/features/webUISharingInternalUsersToRootBlacklisted/shareWithUsers.feature:13]: should cover by
https://github.com/owncloud/web/blob/e1b958326be41975cd53445a6d88bcf92f3be1e2/tests/acceptance/features/webUISharingInternalUsersToRootBlacklisted/shareWithUsers.feature#L58-L67

File: tests/acceptance/features/webUISharingNotifications/notificationLink.feature (hunk 7 above)
Comment (individual-it) [reply tests/acceptance/features/webUISharingNotifications/notificationLink.feature:25]: let the test run and put it into expected to fail list

File: tests/acceptance/features/webUISharingInternalUsersToRoot/shareWithUsers.feature [hunk 12]
```diff
@@ -39,12 +39,10 @@ Feature: Sharing files and folders with internal users
...
-      | lorem.txt  |
-    But these resources should not be listed in the folder "simple-folder" on the webUI
-      | entry_name    |
-      | simple-folder |
+    When the user opens the share dialog for file "simple-folder" using the webUI
+    Then user "Brian Murphy" should be listed as "Owner" in the collaborators list on the webUI
+    When the user opens the share dialog for file "testimage.jpg" using the webUI
+    Then user "Brian Murphy" should be listed as "Owner" in the collaborators list on the webUI
     #    And folder "simple-folder" should be marked as shared by "Brian Murphy" on the webUI
     #    And file "testimage.jpg" should be marked as shared by "Brian Murphy" on the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingInternalUsersToRoot/shareWithUsers.feature:47]: these steps would check if the item is listed on the `sharedWithMePage`. I think we can delete those steps from here

**Review by individual-it — CHANGES_REQUESTED** [review]

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>oc10-integration-notifications</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/web/16406/

<p>


### webUISharingNotificationsToRoot-shareWithUsers-feature-40.png
  


</p></details>

**Review by individual-it — APPROVED** [review]
LGTM @jasson99 are you happy with the current state?

**Comment by jasson99** [comment]
> LGTM @jasson99 are you happy with the current state?

Yes it is okay. What I was confused about and needed to look into the stepsdefinition was in this step `And the user opens the share creation dialog in the webUI`. I felt it was not clear (for me) about opening the share creation dialog for the previously shared resource. I thought it would be clear if the step mentioned that the share creation dialog was opened for the particular share. But once we know what is in the implementation, it might be okay. I think we can merge this if it is understandable to all of us. It was just me being confused then.

**Comment by saw-jan** [comment]
> > LGTM @jasson99 are you happy with the current state?
> 
> @individual-it @saw-jan Yes it is okay. What I was confused about and needed to look into the stepsdefinition was in this step `And the user opens the share creation dialog in the webUI`. I felt it was not clear (for me) about opening the share creation dialog for the previously shared resource. I thought it would be clear if the step mentioned that the share creation dialog was opened for the particular share. But once we know what is in the implementation, it might be okay. I think we can merge this if it is understandable to all of us. It was just me being confused back then.

Yeah, the step continues to work along with above step as the share step doesn't closes the share sidebar. That's why that step is able to open share dailog.

**Pull request merged**